proof-systems/src/darlin/benches/t_dlog_acc_marlin.rs

```diff
@@ -0,0 +1,668 @@
+        || format!(
+            "--->END TIMESTAMP: {:?}",
+            SystemTime::now()
+                .duration_since(UNIX_EPOCH)
+                .unwrap()
+                .as_secs()
+        )
+    );
+}
+
+fn bench_prover_trivial_prev_acc<C1, C2>(
+    group: &mut BenchmarkGroup<WallTime>,
+    num_constraints: usize,
+    segment_size: usize,
+) where
+    C1: ConstraintSynthesizer<<G1 as Group>::ScalarField> + RandomCircuit,
+    C2: ConstraintSynthesizer<<G2 as Group>::ScalarField> + RandomCircuit,
+{
+    let mut rng = OsRng::default();
+
+    let c_g1 = C1::generate_random(num_constraints, &mut rng);
+    let (pc_pk_g1, pc_vk_g1, index_pk_g1, index_vk_g1) =
+        generate_keys::<G1, G2, _>(num_constraints, segment_size, c_g1);
+
+    let c_g2 = C2::generate_random(num_constraints, &mut rng);
+    let (pc_pk_g2, pc_vk_g2, _, index_vk_g2) =
+        generate_keys::<G2, G1, _>(num_constraints, segment_size, c_g2);
+
+    add_to_trace!(
+        || format!("****************{}*******************", num_constraints),
+        || format!(
+            "--->START TIMESTAMP: {:?}",
+            SystemTime::now()
+                .duration_since(UNIX_EPOCH)
+                .unwrap()
+                .as_secs()
+        )
+    );
+
+    group.bench_with_input(
+        BenchmarkId::from_parameter(num_constraints),
+        &num_constraints,
+        |bn, _constraints| {
+            bn.iter_batched(
+                || {
+                    let circ = C1::generate_random(num_constraints, &mut rng);
+
+                    let (dlog_acc, t_acc, bullet_poly, t_poly) = generate_trivial_accumulators(
+                        &pc_pk_g1,
+                        &pc_vk_g1,
+                        &index_vk_g1,
+                        &pc_pk_g2,
+                        &pc_vk_g2,
+                        &index_vk_g2,
+                    );
+
+                    (circ, dlog_acc, t_acc, bullet_poly, t_poly)
+                },
+                |(c, dlog_acc, t_acc, bullet_poly, t_poly)| {
+                    TDLogAccMarlin::<G1, G2, D>::prove(
+                        &index_pk_g1,
+                        &pc_pk_g1,
+                        c,
+                        &t_acc,
+                        &dlog_acc,
+                        &t_poly,
+                        &bullet_poly,
+                        false,
+                        None,
+                    )
+                    .unwrap();
+                },
+                BatchSize::PerIteration,
+            );
+        },
+    );
+    add_to_trace!(
+        || format!("****************{}*******************", num_constraints),
+        || format!(
+            "--->END TIMESTAMP: {:?}",
+            SystemTime::now()
+                .duration_since(UNIX_EPOCH)
+                .unwrap()
+                .as_secs()
+        )
+    );
+}
+
+fn bench_prover_circuit1c_half_segment_size_single_acc(c: &mut Criterion) {
```

Comment:
Is it really worth to create separate bench functions for separate segment sizes ? Sounds like a lot of avoidable code duplication

Reply:
Refactored benches